drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -432,17 +433,84 @@ static int siwx91x_scan(const struct device *dev, struct wifi_scan_params *z_sca
 		return -EINVAL;
 	}
 
-	if (z_scan_config->scan_type == WIFI_SCAN_TYPE_ACTIVE) {
-		sl_scan_config.type = SL_WIFI_SCAN_TYPE_ACTIVE;
-		ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_ACTIVE_SCAN_TIMEOUT,
-						 z_scan_config->dwell_time_active);
+	if (sidev->state == WIFI_STATE_COMPLETED) {
```

Comment:
I wonder if it wouldn't make sense to add a field `int period` to `z_scan_config`. We would start a background scan of `period != 0`. What do you think @jukkar ?

Reply:
There isn't a bgscan support in `struct wifi_scan_params`, and if we want to add it, we need more than just `period`, no?

Reply:
Indeed, we probably also need a couple of threshold values.